### PR TITLE
Change MOAB build process to CMake and turn off blas/lapack & FORTRAN

### LINKED ISSUE #41: add DAGMC to install scripts
Install scripts have been updated to include pyMOAB but not DAGMC.

### PR DESCRIPTION
**Pull request by gonuke**
CMake offers a more robust cross-platform build process for MOAB, particularly for purposes related to PyNE installation.

Also working to remove a BLAS/LAPACK dependency that can be troublesome on some platforms and is unnecessary for purposes in PyNE.

FORTRAN support is not required in MOAB for PyNE use cases.

Also adds DAGMC support, to fix #41

### CODE REVIEW

File: ubuntu_16.04-dev.dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN apt-get install -y --force-yes \
 
 # pyne specific dependencies (excluding python libraries)
 RUN apt-get install -y build-essential git cmake vim emacs gfortran libblas-dev \
-                       python-pip liblapack-dev libhdf5-dev autoconf libtool
+                       python-pip liblapack-dev libeigen3-dev libhdf5-dev autoconf libtool
```
Comment (bam241) [inline ubuntu_16.04-dev.dockerfile:11]: do we still need lapack ?

File: ubuntu_mint.sh [hunk 2]
```diff
@@ -24,10 +24,15 @@ function build_moab {
...
     mkdir -p build
     cd build
-    ../configure --enable-shared --enable-dagmc --enable-pymoab --with-hdf5=$hdf5_libdir --prefix=$install_dir/moab
+    cmake ../ -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_libdir} \
+              -DENABLE_PYMOAB=ON \
+              -DENABLE_BLASLAPACK=OFF \
+              -DENABLE_FORTRAN=OFF \
+              -DCMAKE_INSTALL_PREFIX=${install_dir}/moab \
+              -DCMAKE_C_COMPILER=${CC} \
+              -DCMAKE_CXX_COMPILER=${CXX}          -D
```
Comment (bam241) [inline ubuntu_mint.sh:35]:
```suggestion
              -DCMAKE_CXX_COMPILER=${CXX}
```

File: ubuntu_16.04-dev.dockerfile (hunk 1 above)
Comment (gonuke) [reply ubuntu_16.04-dev.dockerfile:11]: For DAGMC (for now)

**Comment by bam241** [comment]
I am testing the docker build for stable and dev.

this looks good to me, let me know when you done with it :)

**Comment by bam241** [comment]
got an error with the ubuntu.18.04_stable (docker):
```

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
MOAB_LIBRARIES_STATIC
    linked by target "dagmc-static" in directory /root/DAGMC/src/dagmc
    linked by target "pyne_dagmc-static" in directory /root/DAGMC/src/pyne

-- Configuring incomplete, errors occurred!
```

**Comment by bam241** [comment]
I think we are missing the MOAB STATIC.
this is why we build 2 time MOAB in [here](https://github.com/pyne/pyne/blob/develop/docker/ubuntu_18.04-dev.dockerfile)
(this build process to get both shared and static version of the MOAB lib is not clean... )

**Comment by bam241** [comment]
still have the problem.
I think we need to build MOAB twice...

**Comment by gonuke** [comment]
I don't think we need MOAB static.  I had a typo in the MOAB build line.  Trying again with that fixed.

**Comment by gonuke** [comment]
Hmmm.... that's not it - still investigating.

**Comment by bam241** [comment]
I think we have an issue with DAGMC cmake, we whatever reason we are looking for both.

**Comment by bam241** [comment]
(we build both in the pyne docker image for CI)

**Comment by gonuke** [comment]
I don't think we need to - at least not anymore.

**Comment by bam241** [comment]
DAGMC is building !

**Comment by gonuke** [comment]
But PyNE can't find pymoab.... still needs some work :(

**Comment by bam241** [comment]
```
--    HDF5 Include directory: /usr/include/hdf5/serial
--    HDF5 Library directories:
--    HDF5 Libraries: /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so;/usr/lib/x86_64-linux-gnu/libpthread.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so
-- MOAB Root: /root/opt/moab
-- Found MOAB: /root/opt/moab/lib/libMOAB.so
-- MOAB header files: /root/opt/moab/include
-- MOAB library: /root/opt/moab/lib/libMOAB.so
-- MOAB Found: TRUE
-- DAGMC Root: /root/opt/dagmc
```
MOAB is found by PyNE  when I build on `ubuntu.18.04.stable` (docker)

**Comment by gonuke** [comment]
It finds it to build, but can't find `pymoab` during testing

**Comment by gonuke** [comment]
Because I don't enable pymoab!!!

**Comment by gonuke** [comment]
I just tested all of these and they all appear to work

**Comment by kkiesling** [comment]
before this gets merged, can you switch it to the `main` branch

**Comment by kkiesling** [comment]
never mind- turns out I can change the PR base branch myself!

**Comment by kkiesling** [comment]
@bam241 @gonuke - is this ready to be merged?

**Comment by bam241** [comment]
there is conflict now :(

@kkiesling I did not test it yet, I would like to run it  (we could actually setup CI for this... ) 
I'll merge it after I try it

**Comment by kkiesling** [comment]
I resolved the conflict online here, merging now.

**Comment by kkiesling** [comment]
oh wait, not merging since you want to test. I will wait.

**Comment by bam241** [comment]
doing to docker tests:
- [ ] 16.04 dev
- [ ] 16.04 stable
- [ ] 18.04 dev
- [ ] 18.04 stable

**Comment by bam241** [comment]
` --force-yes` seems to be now deprecated in ubuntu `18.04`, it is actually not required. so we can as well remove it.

**Comment by bam241** [comment]
Everything is working.
Merging!
thank you @gonuke !

**Pull request merged**